regulations/static/regulations/js/source/views/main/child-view.js

```diff
@@ -151,15 +151,19 @@ var ChildView = Backbone.View.extend({
             var url = this.url,
                 hashPosition;
 
+            console.log(options);
+
             // if a hash has been passed in
             if (options && typeof options.scrollToId !== 'undefined') {
                 url += '#' + options.scrollToId;
-            }
-            else {
+            } else {
                 hashPosition = (typeof Backbone.history.fragment === 'undefined') ? -1 : Backbone.history.fragment.indexOf('#');
                 //  Be sure not to lose any hash info
                 if (hashPosition !== -1) {
-                    url += Backbone.history.fragment.substr(hashPosition);
+                    url = url.slice(0, hashPosition) + '#' + options.id;
```

Comment:
If you go with that approach, maybe take it one step further and remove the nested if:

```javascript
if (options && typeof options.scrollToId !== 'undefined') {
    url += '#' + options.scrollToId;
// Don't lose the hash info.
} else if (typeof Backbone.history.fragment !== 'undefined') {
    hashPosition =  Backbone.history.fragment.indexOf('#');
    url = url.slice(0, hashPosition) + '#' + options.id;
} else if (options.type !== 'diff')  {
    url += '#' + options.id;
}
```

It's a bit worry-some, though, that there's no default.